wsproto/connection.py

```diff
@@ -275,6 +275,11 @@ def events(self):
                 if frame.opcode is Opcode.PING:
                     assert frame.frame_finished and frame.message_finished
                     self._outgoing += self._proto.pong(frame.payload)
+                    yield PingReceived(frame.payload)
+
+                elif frame.opcode is Opcode.PONG:
+                    assert frame.frame_finished and frame.message_finished
+                    yield PongReceived(frame.payload)
```

Comment:
Well, arguably this is difficult. I couldn't find any [errata in the RFC](https://www.rfc-editor.org/errata_search.php?rfc=6455) (yet).

IMO this sentence in the RFC is pretty clear:
> A Pong frame sent in response to a Ping frame must have identical
   "Application data" as found in the message body of the Ping frame
   being replied to.
Strangely it doesn't make use of the usual "MUST HAVE" in capital letters to indicate an absolute requirement / condition...